src/Corax/IndexWriter.cs

```diff
@@ -405,35 +532,38 @@ public unsafe long Update(string field, Span<byte> key, LazyStringValue id, Span
             return entryId;
         }
         
+       // idInTree - encoded with frequency and container type
         public long Update(string field, Span<byte> key, LazyStringValue id, Span<byte> data, ref long numberOfEntries)
         {
-            if (TryGetEntryTermId(field, key, out var entryId) == false)
+            if (TryGetEntryTermId(field, key, out var idInTree) == false)
             {
                 numberOfEntries++;
                 return Index(id, data);
             }
             // if there is more than a single entry for this key, delete & index from scratch
             // this is checked by calling code, but cheap to do this here as well.
-            if((entryId & (long)TermIdMask.EnsureIsSingleMask) != 0)
+            if((idInTree & (long)TermIdMask.EnsureIsSingleMask) != 0)
```

Comment:
Shouldn't you remove the frequency before making this comparision?

Reply:
No, since the container type is saved on the two least significant bits. So this is ok.